### PR TITLE
[Feature] Create product details page

### PR DESCRIPTION
**Pull request by oreoluwade**
#### What does this PR do?
- Creates a product-details page

#### Description of Task to be completed?
- Create a product-details component with the accompanying route
- Activate navigation and display of available details on the product-details page.
- Abstract product component from product-list component.

#### How should this be manually tested?
- Run the application
- Click on the `cosmetics` link
- Click on any of the products
- You should be navigated to a page that contains details about only the product with the url showing the ID of the particular product.

#### What are the relevant pivotal tracker stories?
- [#166755827](https://www.pivotaltracker.com/story/show/166755827)

#### Screenshots (if appropriate)
Screen Shot 2019-06-18 at 4 01 58 PM (2)

#### Extra details
- When the product-details page is reloaded, there is a redirect to `products`. This is due to the fact that the product details is only passed from one route to the other as it stands. Shortly, a story is coming to write the data to cache so that data persistence is in place once redirected to a details page.
- The product-details page is minimalistic because my first goal is to make the page/component available and then work (and more details) can be done on it subsequently.

### CODE REVIEW

**Comment by oreoluwade** [comment]
> The spacing between the products vanished. Is this deliberate?

The spacing between them in the grid? Hmmm. Let me investigate that. Not sure I touched the styling for pre-existing pages.

File: src/components/Product/product-details.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    height: 200
+  },
+  text: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginLeft: '8rem'
+  }
+});
+
+function ProductDetails({ location, classes }) {
```
Comment (omorilewa) [inline src/components/Product/product-details.jsx]: Why are we using `function` ?

Comment (oreoluwade) [reply src/components/Product/product-details.jsx]: Lol. We have always alternated between arrow function and using the function keyword for functional components. Case in point is the `App` component. However, we could stick with one if we so desire. I could refactor this one to be an arrow functional component.

**Comment by oreoluwade** [comment]
> The spacing between the products vanished. Is this deliberate?

@fisayomi Made an update regarding the spacing. Please check if suitable or not.

File: src/components/Product/product.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    <GridListTile key={product.name} cols={1}>
+      <Card className={classes.card}>
+        <Link to={location} className={classes.link}>
+          <CardActionArea className={classes.card}>
+            <img
+              src={product.imageUrl}
+              alt={product.name}
+              className={classes.image}
+            />
+            <CardContent>
```
Comment (fisayomi) [inline src/components/Product/product.jsx]: Can you make CardContent component boolean? It shouln't be available when in products-detail page

**Comment by fisayomi** [comment]
> > The spacing between the products vanished. Is this deliberate?
> 
> @fisayomi Made an update regarding the spacing. Please check if suitable or not.

It's better. Thanks.

File: src/components/Product/product.jsx (hunk 2 above)
Comment (oreoluwade) [reply src/components/Product/product.jsx]: I could do that. However, it's quite possible the product details page won't even look the way it looks in the end. For instance, the image could be bigger, the location of the name and description could be different, the price could be included, different views of the image could be included. The present structure is to have something displayed first of all. If you still feel strongly about it, I could still yank those details off, or prevent them from showing if the `match.params.productId` is present in the uri. Expecting your response.

Comment (fisayomi) [reply src/components/Product/product.jsx]: I feel like when we navigate to the Product detail page, The pic should not contain any info on it. We would however work on zoom effects for the image

Comment (oreoluwade) [reply src/components/Product/product.jsx]: Okay I get. But the product details page does not contain the same `CardContent` component on the ProductList page though. I have made some changes, but could you have a look again at the files in question? The Product component and the ProductDetails component?

Comment (fisayomi) [reply src/components/Product/product.jsx]: Oh, we should make use of the same card content I guess, That can be done in another story. Looks better now. Thank you.

**Pull request merged**